### PR TITLE
Added jwt time to live with env parameter

### PR DESCRIPTION
**Pull request by vantihovich**

### CODE REVIEW

File: swagger/handlers/users.go [hunk 1]
```diff
@@ -199,15 +200,18 @@ func (h *UsersHandler) UserLogin(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	//jwtExpirationTime := time.Now().Local().Add(time.Minute * time.Duration(h.jwtParam.JwtTTL))
```
Comment (apaliavy) [inline swagger/handlers/users.go:203]: commented code

File: swagger/middleware/middleware.go [hunk 2]
```diff
@@ -26,16 +27,20 @@ func Authorize(cfg string, f http.HandlerFunc) http.HandlerFunc {
...
 				w.WriteHeader(http.StatusUnauthorized)
 				return
 			}
 			log.WithError(err).Info("Error parsing the token")
 			w.WriteHeader(http.StatusBadRequest)
 			return
 		}
+
 		if !tkn.Valid {
+			fmt.Println("check condition2")
```
Comment (apaliavy) [inline swagger/middleware/middleware.go:43]: debug statement

File: swagger/middleware/middleware.go [hunk 3]
```diff
@@ -26,16 +27,20 @@ func Authorize(cfg string, f http.HandlerFunc) http.HandlerFunc {
 		tkn, err := jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
 			return []byte(cfg), nil
 		})
+
 		if err != nil {
-			if err.Error() == jwt.ErrSignatureInvalid.Error() {
+			if err == jwt.ErrSignatureInvalid {
+				fmt.Println("check condition")
```
Comment (apaliavy) [inline swagger/middleware/middleware.go:33]: debug statement

File: swagger/handlers/users.go [hunk 4]
```diff
@@ -199,15 +200,18 @@ func (h *UsersHandler) UserLogin(w http.ResponseWriter, r *http.Request) {
...
+
 	claims := &claims{
 		UserID: user.ID,
 		StandardClaims: jwt.StandardClaims{
-			Issuer: "AuthService",
+			Issuer:    "AuthService",
+			ExpiresAt: time.Now().Local().Add(time.Minute * time.Duration(h.jwtParam.JwtTTL)).Unix(),
```
Comment (apaliavy) [inline swagger/handlers/users.go:209]: any reasons to use .Local() here?

File: swagger/handlers/users.go (hunk 1 above)
Comment (vantihovich) [reply swagger/handlers/users.go:203]: removed

File: swagger/middleware/middleware.go (hunk 2 above)
Comment (vantihovich) [reply swagger/middleware/middleware.go:43]: removed

File: swagger/middleware/middleware.go (hunk 3 above)
Comment (vantihovich) [reply swagger/middleware/middleware.go:33]: removed

File: swagger/handlers/users.go (hunk 4 above)
Comment (vantihovich) [reply swagger/handlers/users.go:209]: at this time no reasons, removed

File: swagger/handlers/users.go [hunk 5]
```diff
@@ -202,12 +203,13 @@ func (h *UsersHandler) UserLogin(w http.ResponseWriter, r *http.Request) {
...
 		StandardClaims: jwt.StandardClaims{
-			Issuer: "AuthService",
+			Issuer:    "AuthService",
+			ExpiresAt: time.Now().Add(time.Minute * time.Duration(h.jwtParam.JwtTTL)).Unix(),
 		},
 	}
 
 	jwToken := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
-	response.Token, err = jwToken.SignedString([]byte(h.jwtParam))
+	response.Token, err = jwToken.SignedString([]byte(h.jwtParam.SecretKey))
```
Comment (apaliavy) [inline swagger/handlers/users.go:212]: Actually, I think we might want to move the whole JWT logic into a separate package. 
So in login handler you'd have something very simple like :
```
if !user.Active {
 // user active check here 
}

response.Token, err = jwt.GetToken(user.ID)
```

File: swagger/middleware/middleware.go [hunk 6]
```diff
@@ -26,15 +26,17 @@ func Authorize(cfg string, f http.HandlerFunc) http.HandlerFunc {
 		tkn, err := jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
 			return []byte(cfg), nil
 		})
+
 		if err != nil {
-			if err.Error() == jwt.ErrSignatureInvalid.Error() {
+			if err == jwt.ErrSignatureInvalid {
```
Comment (apaliavy) [inline swagger/middleware/middleware.go:31]: use `errors.Is() or errors.As()`

Comment (vantihovich) [reply swagger/middleware/middleware.go:31]: fixed

File: swagger/handlers/users.go (hunk 5 above)
Comment (vantihovich) [reply swagger/handlers/users.go]: done

**Review by apaliavy — APPROVED** [review]

**Pull request merged**